funcx_web_service/routes/funcx.py

```diff
@@ -704,26 +709,42 @@ def get_ep_stats(user: User, endpoint_id):
 
     rc = g_redis_client()
 
-    status = {"status": "offline", "logs": []}
+    status: str = "offline"
+    status_logs: t.List[t.Dict[str, t.Any]] = []
     try:
         end = min(rc.llen(f"ep_status_{endpoint_id}"), last)
         print("Total len :", end)
         items = rc.lrange(f"ep_status_{endpoint_id}", 0, end)
         if items:
             for i in items:
-                status["logs"].append(json.loads(i))
+                dataitem = json.loads(i)
+                if not isinstance(dataitem, dict):
+                    raise EndpointStatsError(
+                        endpoint_id, "endpoint stats failed to load, nondict data"
+                    )
+                status_logs.append(dataitem)
```

Comment:
This is mostly a sidenote and I'm not going to hold up the PR over it, but I'm not a fan of this kind of inline type checking to handle mypy limitations.  This breaks up the readability of the function just to handle type checking that really could be done elsewhere, or may not even be necessary.  The original `mypy` complaint here was `Sequence[str] has no atribute "append"`.  However, from the former definition of `status = {"status": "offline", "logs": []}` it's easily verifiable that `status["logs"].append` must exist.  I personally find it far more readable in those cases where mypy is unable to appropriately deduce the type to just ignore mypy  (e.g. `# type: ignore`) and still get some static safety where type really is ambiguous.  

That said, having mypy is better than not and minimizing changes trumps major refactoring for this purpose, so if this is the simplest fix so be it.  Longer term I'd prefer to see things like use of Pydantic, jsonschema or expicit data structure validation rather than interrupting readability.

Reply:
I would actually call the original code much _less_ readable, with respect to the line in question.

Rather than constructing a list and a string, and then assembling them into a dict at the end of the function, this constructed a dict and then referred to it's members. It may be shorter, but it's worse.
1. The ultimate contents of the constructed dict are unclear, as it is modified within the loop scope. You don't trivially know if elements are added or even removed.
2. Instead of a variable name, you're using a string to refer to a specific value (and write that value). A typo when running `__setitem__` is valid but undesirable. A typo when setting a variable can often be caught in linting as an unused variable.

There are cases in which a type ignore is appropriate. Those are exceedingly rare. More often, the right thing may be to cast to the desired type.

We can debate various cases, but I think this is a very clear cut one in which a type ignore is not desirable.

I'm sure this will come up again in a more meaningful context, and we can discuss when it does. Just sharing an alternate perspective.